js/shared.js

```diff
@@ -395,6 +395,80 @@ function undoBrowserSelection(s){
     s?.removeAllRanges?.() ?? s?.empty?.()
 }
 
+/** 
+ * Discover proper identifier for manuscript or suggest new IRI. 
+ *
+ * @async
+ * @function findShelfmark
+ * @param { string } msid Identifier to match or validate.
+ * @param { boolean } [forceNew] True to skip search for collision. *risky*
+ * @returns Valid msid string or parent IRI
+ */ 
+async function findShelfmark(msid, forceNew) {
+    try {
+        // wash msid
+        if (typeof msid !== 'string') {
+            const invalidInputEvent = new CustomEvent("Failed to Query Rerum. Invalid msid identifier input.")
+            globalFeedbackBlip(invalidInputEvent, 'Failed to query for Shelfmark: Attempted to add a non string.', false)
+            return
+        }
+
+        const cleanMsid = msid.replace(/[@$%*?]+/g, '') 
+        .replace(/\s+/g, ' ') 
+        .trim() 
+
+        if (cleanMsid.length === 0) {
+            const invalidInputEvent = new CustomEvent("Failed to Query Rerum. Invalid msid identifier input.")
+            globalFeedbackBlip(invalidInputEvent, 'Failed to query for Shelfmark: Attempted to add an empty string.', false)
+            return
+        }
+
+        // return washed msid or parent msid
+        if (forceNew) {
+            return cleanMsid
+        }
+        else {
+            // check for parent msid
+            const query = {
+                "body.alternateTitle.value": cleanMsid,
+                "__rerum.generatedBy" : __constants.generator
+            }
+    
+            const annotation = await fetch(`${__constants.tiny+"/query"}`, {
+                method: 'POST',
+                mode: 'cors',
+                headers: {
+                    "Content-Type": "application/json;charset=utf-8"
+                },
+                body: JSON.stringify(query)
+            })
+            .then(resp => resp.json())
+            .catch(err => {
+                console.error(err)
+                const qryFail = new CustomEvent("Failed to query RERUM.")
+                globalFeedbackBlip(qryFail, 'Failed to find annotation with msid identifier.', false)
+                return
+            })
+    
+            if(annotation.length > 0){
+                return annotation[0]["target"]
+            }
+            else {
```

Comment:
You have a `return` right before this, so the `else` is extra.
Also, "Something went wrong" can be reached if there are correctly no annotations found (result = `[]`). This doesn't feel like a queryFail - like getting a 500 on a 404.

Reply:
In cases where the result is [], I changed it from an error blip to just returning null in cases where the query finds nothing.